### PR TITLE
Adds back reverted commits and fixes NPE in rolling-update cmd

### PR DESCRIPTION
**Pull request by davidxia**

### CODE REVIEW

**Comment by mattnworb** [comment]
between this PR, and #1181 #1183 #1184 I am not seeing any explanation of what caused the exceptions originally, so it is really hard to review this at all.

What caused the NullPointerExceptions in 1181?
Why do tests not catch that? Can tests be added that catch that?
Why revert it, and why now revert the revert without additional changes?

**Comment by davidxia** [comment]
> What caused the NullPointerExceptions in 1181?

https://github.com/spotify/helios/pull/1181/files didn't cause any NPEs. In fact, it fixes a bunch of server-side NPEs.

> Why do tests not catch that? Can tests be added that catch that?

Are you talking about catching the NPE fixed by #1181?

> Why revert it, and why now revert the revert without additional changes?

Not sure about the first. For the second question, there are additional changes. I added a new commit.

File: helios-tools/src/main/java/com/spotify/helios/cli/command/RollingUpdateCommand.java [hunk 1]
```diff
@@ -184,27 +186,36 @@ protected int runWithJobId(final Namespace options, final HeliosClient client,
       return 1;
     }
 
+    final RolloutOptions optionsFromJob = job.getRolloutOptions();
```
Comment (davidxia) [inline helios-tools/src/main/java/com/spotify/helios/cli/command/RollingUpdateCommand.java]: NPE starts here. This variable can be null.

File: helios-tools/src/main/java/com/spotify/helios/cli/command/RollingUpdateCommand.java [hunk 2]
```diff
@@ -21,16 +21,17 @@
 package com.spotify.helios.cli.command;
 
 import static com.google.common.base.Preconditions.checkArgument;
-import static com.spotify.helios.common.descriptors.Job.EMPTY_TOKEN;
 import static java.lang.String.format;
 import static net.sourceforge.argparse4j.impl.Arguments.storeTrue;
 
+import com.google.api.client.repackaged.com.google.common.base.Objects;
```
Comment (mattnworb) [inline helios-tools/src/main/java/com/spotify/helios/cli/command/RollingUpdateCommand.java]: looks like an accidental import

Comment (davidxia) [reply helios-tools/src/main/java/com/spotify/helios/cli/command/RollingUpdateCommand.java]: thanks, fixed

**Comment by caipre** [comment]
👍

**Pull request merged**